src/ImageViewer.cpp

```diff
@@ -1475,6 +1501,49 @@ void ImageViewer::setGamma(float value) {
     mImageCanvas->setGamma(value);
 }
 
+void ImageViewer::showCaptionEditTextBox() {
+    if (!mCurrentImage || mCurrentCaptionTextBox) {
+        return;
+    }
+
+    int id = imageId(mCurrentImage);
+    mImageButtonContainer->child_at(id)->set_visible(false);
+
+    mCurrentCaptionTextBox = new TextBox{nullptr, ""};
+    mCurrentCaptionTextBox->set_editable(true);
+    mCurrentCaptionTextBox->set_alignment(TextBox::Alignment::Left);
+    mCurrentCaptionTextBox->set_placeholder("Enter caption");
+    mCurrentCaptionTextBox->set_callback([this](const string& name) {
+        if (!mCurrentImage) {
+            return true;
+        }
+
+        int id = imageId(mCurrentImage);
+        ImageButton* ib = dynamic_cast<ImageButton*>(mImageButtonContainer->children()[id+1]);
```

Comment:
This is brittle and prone to errors if tev keeps loading and/or closing images while a caption is being edited. I think it's hopeless to catch all those cases and make them close the textbox -- instead, caption editing should be robust to changes in the images (even to the image currently being edited getting closed).

If I were to tackle this, I'd make the text box a child of the image button in question, moving much of the logic into `ImageButton.cpp` and `ImageButton.h`. That way, the text box automatically disappears if the image does, and the caption can be edited without having to hold a reference to the image itself. (This would also make my above comment obsolete.)